File: 2418-sort-the-people/2418-sort-the-people.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function(names, heights) {
    let obj = {};
    let results = [];
    for(let i = 0; i < heights.length; i++) {
        obj[heights[i]] = names[i];
    }
    const arr = Object.entries(obj);
    arr.sort((a, b) => b[0] - a[0]);
    for(let i = 0; i < arr.length; i++) {
        results.push(arr[i][1]);
    }
    return results;
};

File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        results = []
        list_people = list(zip(names, heights))
        list_people.sort(key=lambda x: -x[1])
        for name, height in list_people:
            results.append(name)
        return results